tests/cli-ui.rs

```diff
@@ -0,0 +1,18 @@
+use rustup::utils::utils;
+
+#[test]
+fn ui_tests() {
+    let t = trycmd::TestCases::new();
+    let rustup_init = trycmd::cargo::cargo_bin("rustup-init");
+    let rustup = trycmd::cargo::cargo_bin("rustup");
+    t.register_bin("rustup-init", &rustup_init);
+    // Copy rustup-init to rustup so that the tests can run it.
+    utils::copy_file(&rustup_init, &rustup).unwrap();
```

Comment:
ah yes here; this can just use `fs::copy`, it doesn't need all the heuristics and UI glue that we have in utils.